### ISSUE TITLE
chmod: cannot access './var': No such file or directory

### ISSUE THREAD

**Issue by blopa** [clean-docker/Magento2#20]
Hello Rafael!

First of all, thanks for this docker project, it will surely help me in many ways I can't even imagine :)

Sadly, I'm having a hard time getting the container to properly work. When I try to run `install-magento2` I got this error message:

```
root@1500be452086:/var/www/html# install-magento2
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Installing magento/project-community-edition (2.2.1)
  - Installing magento/project-community-edition (2.2.1): Loading from cache
Created project in .
Loading composer repositories with package information
Updating dependencies (including require-dev)
/usr/local/bin/install-magento2: line 6:    76 Killed                  composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition .
chmod: cannot access './var': No such file or directory
chmod: cannot access './vendor': No such file or directory
chmod: cannot access './pub/static': No such file or directory
chmod: cannot access './pub/media': No such file or directory
chmod: cannot access './app/etc': No such file or directory
mv: cannot stat 'package.json.sample': No such file or directory
mv: cannot stat 'Gruntfile.js.sample': No such file or directory
```

What am I doing wrong? I'm using DigitalOcean's one-click apps for Docker on Ubuntu 16.04. https://www.digitalocean.com/products/one-click-apps/docker/

Best.

**Comment by rafaelstz** [comment]
Hi @blopa, it's happening when the script tries to install using the database username as *magento*, so open your installation in your browser and run the install process manually to setup using your database name, then open your `app/etc/env.php` and change like this sample below, **changing the host, database and username like it's in the original file.**

```php
<?php
return array(
    'backend' =>
        array(
            'frontName' => 'admin',
        ),
    'crypt' =>
        array(
            'key' => '5e56dc6d8f24f711bb4f289cca1beda9',
        ),
    'session' =>
        array(
            'save' => 'redis',
            'redis' =>
                array(
                    'host' => 'redis-session',
                    'port' => '6379',
                    'password' => '',
                    'timeout' => '2.5',
                    'persistent_identifier' => '',
                    'database' => '2',
                    'compression_threshold' => '2048',
                    'compression_library' => 'gzip',
                    'log_level' => '1',
                    'max_concurrency' => '6',
                    'break_after_frontend' => '5',
                    'break_after_adminhtml' => '30',
                    'first_lifetime' => '600',
                    'bot_first_lifetime' => '60',
                    'bot_lifetime' => '7200',
                    'disable_locking' => '0',
                    'min_lifetime' => '60',
                    'max_lifetime' => '2592000'
                )
        ),
    'db' =>
        array(
            'table_prefix' => '',
            'connection' =>
                array(
                    'default' =>
                        array(
                            'host' => 'db',
                            'dbname' => 'magento',
                            'username' => 'magento',
                            'password' => 'magento',
                            'active' => '1',
                        ),
                ),
        ),
    'resource' =>
        array(
            'default_setup' =>
                array(
                    'connection' => 'default',
                ),
        ),
    'x-frame-options' => 'SAMEORIGIN',
    'MAGE_MODE' => 'production',
    'cache_types' =>
        array(
            'config' => 1,
            'layout' => 1,
            'block_html' => 1,
            'collections' => 1,
            'reflection' => 1,
            'db_ddl' => 1,
            'eav' => 1,
            'customer_notification' => 1,
            'full_page' => 0,
            'config_integration' => 1,
            'config_integration_api' => 1,
            'translate' => 1,
            'config_webservice' => 1,
            'compiled_config' => 1,
        ),
    'install' =>
        array(
            'date' => 'Thu, 05 Jan 2017 22:49:50 +0000',
        ),
    'cache' =>
        array(
            'frontend' =>
                array(
                    'default' =>
                        array(
                            'backend' => 'Cm_Cache_Backend_Redis',
                            'backend_options' =>
                                array(
                                    'server' => 'redis',
                                    'port' => '6379',
                                    'persistent' => '',
                                    'database' => '0',
                                    'password' => '',
                                    'force_standalone' => '0',
                                    'connect_retries' => '1',
                                    'read_timeout' => '10',
                                    'automatic_cleaning_factor' => '0',
                                    'compress_data' => '1',
                                    'compress_tags' => '1',
                                    'compress_threshold' => '20480',
                                    'compression_lib' => 'gzip',
                                    'use_lua' => '0',
                                ),
                        ),
                    'page_cache' =>
                        array(
                            'backend' => 'Cm_Cache_Backend_Redis',
                            'backend_options' =>
                                array(
                                    'server' => 'redis',
                                    'port' => '6379',
                                    'persistent' => '',
                                    'database' => '1',
                                    'password' => '',
                                    'force_standalone' => '0',
                                    'connect_retries' => '1',
                                    'lifetimelimit' => '57600',
                                    'compress_data' => '0',
                                ),
                        ),
                ),
        ),
);
```

You can use these commands below to finish:

```bash
php -f ./bin/magento indexer:set-mode schedule \
&& php -f ./bin/magento sampledata:deploy \
&& composer update -vvv \
&& php -f ./bin/magento setup:upgrade \
&& php -f ./bin/magento indexer:reindex \
&& php -f ./bin/magento cache:flush \
&& php -f ./bin/magento deploy:mode:set production \
&& composer dump-autoload -o ;
mv package.json.sample package.json ; mv Gruntfile.js.sample Gruntfile.js \
&& npm i
```

If you have any question, please let me know!
Thanks for your feedback, I'm glad that you liked it 😄

**Comment by blopa** [comment]
Hey Rafael.

Could you ELI5 that for me? Should I try to run the install via browser after or before running `install-magento2`? Because after all I have in my `/var/www/html/` container folder is a `index.php` file, and after running `install-magento2` I got the same error as before, and when I try to access my site via browser I get a 403.

I would really appreciate if you could show me the step by step of how to get it up and running.

Best.

**Comment by rafaelstz** [comment]
Sure, before doing the steps above you can run:

```bash
rm index.php
composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition . ;
chmod u+w -R ./var ./vendor ./pub/static ./pub/media ./app/etc
chmod u+x ./bin/magento
```

Then you open your browser....

If you have any question, please let me know!

**Comment by blopa** [comment]
Hi. So I trie running like you said, but it seems that something is killing my `composer` command, because all I got in the end was a update folder and a json file:

[weird](https://i.imgur.com/Bc1Gk10.png)

So I just got the Magento2 files using `wget` and I got it running, but I can't get pass the Readiness Check:

[weird2](https://i.imgur.com/LP3Sr2Q.png)

I tried running `chmod u+w -R ./var ./vendor ./pub/static ./pub/media ./app/etc` and `chmod u+x ./bin/magento` but no luck

**Comment by rafaelstz** [comment]
hmmm strange, but in your install process use **db** instead  **localhost**, and in your docker-compose.yml change the **PMA_USER** and **PMA_PASSWORD**, and run `docker-compose stop && docker-compose up -d`, to use your new user and password.

<img width="497" alt="screen shot 2017-11-11 at 03 59 29" src="https://user-images.githubusercontent.com/610598/32686845-45a463ae-c695-11e7-8ab5-5b0c3602e185.png">

**Comment by blopa** [comment]
I've deleted my previous host image and start from scratch using `db` and couldn't get passed Step 6. Almost in the end it gets a mysql exception, and after data I can't access phpmyadmin like before - Access denied.

I feel so useless, what am I doing wrong?

**Comment by rafaelstz** [comment]
Nvm @blopa , you aren't doing wrong, until today I didn't need to implement into the real server, so I'm not soo familiar with this issues. BTW, could you provide me the issues print screen?
